src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -27636,6 +27710,358 @@ void Main(string? s)
+                // (22,13): warning CS8601: Possible null reference assignment.
+                //         x = default; // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(22, 13),
+                // (32,16): warning CS8603: Possible null reference return.
+                //         return default; // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "default").WithLocation(32, 16)
+                );
+        }
+
+        [Fact]
+        [WorkItem(36039, "https://github.com/dotnet/roslyn/issues/36039")]
+        public void UnconditionalAttributesAffectPropertyBodies_ReferenceType()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [DisallowNull] string? P1
+    {
+        get { return null; }
+        set { value.ToString(); }
+    }
+
+    [AllowNull] string P2
+    {
+        get { return null; } // 1
+        set { value.ToString(); } // 2
+    }
+
+    [MaybeNull] string P3
+    {
+        get { return null; }
+        set { value.ToString(); }
+    }
+
+    [NotNull] string? P4
+    {
+        get { return null; } // 3
+        set { value.ToString(); } // 4
+    }
+}
+";
+            var comp = CreateCompilation(
+                new[] { source, DisallowNullAttributeDefinition, AllowNullAttributeDefinition, NotNullAttributeDefinition, MaybeNullAttributeDefinition },
+                options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (13,22): warning CS8603: Possible null reference return.
+                //         get { return null; } // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(13, 22),
+                // (14,15): warning CS8602: Dereference of a possibly null reference.
+                //         set { value.ToString(); } // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "value").WithLocation(14, 15),
+                // (25,22): warning CS8603: Possible null reference return.
+                //         get { return null; } // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(25, 22),
+                // (26,15): warning CS8602: Dereference of a possibly null reference.
+                //         set { value.ToString(); } // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "value").WithLocation(26, 15)
+                );
+        }
+
+        [Fact]
+        [WorkItem(36039, "https://github.com/dotnet/roslyn/issues/36039")]
+        public void UnconditionalAttributesAffectPropertyBodies_ReferenceType_AutoProp()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [DisallowNull] string? P1 { get; set; } = """";
+    [AllowNull] string P2 { get; set; } = """";
+    [MaybeNull] string P3 { get; set; } = """";
+    [NotNull] string? P4 { get; set; } = """";
```

Comment:
It's not clear we're that the attributes affect the property bodies since there are no warnings. Should the assignments be changed to ` = null'` to at least test two of the four?